ci-operator/step-registry/openstack/provision/vhostuser/openstack-provision-vhostuser-commands.sh

```diff
@@ -0,0 +1,158 @@
+            name=$(echo "${i}" | awk '{print $1}')
+            updating=$(echo "${i}" | awk '{print $4}')
+            if [[ "${updating}" == "True" ]]; then
+                UPDATING=true
+            else
+                echo "Waiting for mcp ${name} to start rolling out"
+                UPDATING=false
+            fi
+        done <<< "$(oc get mcp worker --no-headers)"
+
+        if [[ "${UPDATING}" == "true" ]]; then
+            echo "Workers are rolling out"
+            return 0
+        else
+            sleep "${INTERVAL}"
+            CNT=$((CNT))-1
+        fi
+
+        if [[ $((CNT)) -eq 0 ]]; then
+            echo "Workers did not successfully start rolling out"
+            oc get mcp "${name}"
+            return 1
+        fi
+    done
+}
+
+function check_workers_updated() {
+    INTERVAL=60
+    CNT=20
+
+    while [ $((CNT)) -gt 0 ]; do
+        READY=false
+        while read -r i
+        do
+            name=$(echo "${i}" | awk '{print $1}')
+            updated=$(echo "${i}" | awk '{print $3}')
+            updating=$(echo "${i}" | awk '{print $4}')
+            degraded=$(echo "${i}" | awk '{print $5}')
+            degraded_machine_cnt=$(echo "${i}" | awk '{print $9}')
+
+            if [[ "${updated}" == "True" && "${updating}" == "False" && "${degraded}" == "False" && $((degraded_machine_cnt)) -eq 0 ]]; then
+                READY=true
+            else
+                echo "Waiting for mcp ${name} to rollout"
+                READY=false
+            fi
+        done <<< "$(oc get mcp worker --no-headers)"
+
+        if [[ "${READY}" == "true" ]]; then
+            echo "Workers have successfully rolled out"
+            return 0
+        else
+            sleep "${INTERVAL}"
+            CNT=$((CNT))-1
+        fi
+
+        if [[ $((CNT)) -eq 0 ]]; then
+            echo "Workers did not successfully roll out"
+            oc get mcp "${name}"
+            return 1
+        fi
+    done
+}
+
+export OS_CLIENT_CONFIG_FILE="${SHARED_DIR}/clouds.yaml"
+
+if [[ ${OPENSTACK_PERFORMANCE_NETWORK} == "" ]]; then
+    echo "OPENSTACK_PERFORMANCE_NETWORK is not set"
+    exit 1
+fi
+
+NETWORK_ID=$(openstack network show "${OPENSTACK_PERFORMANCE_NETWORK}" -f value -c id)
+if [[ "${NETWORK_ID}" == "" ]]; then
+    echo "Failed to find network ${OPENSTACK_PERFORMANCE_NETWORK}"
+    exit 1
+fi
+
+# For disconnected or otherwise unreachable environments, we want to
+# have steps use an HTTP(S) proxy to reach the API server. This proxy
+# configuration file should export HTTP_PROXY, HTTPS_PROXY, and NO_PROXY
+# environment variables, as well as their lowercase equivalents (note
+# that libcurl doesn't recognize the uppercase variables).
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+	# shellcheck disable=SC1090
+	source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+SCRIPT_BASE64=$(curl --retry 10 https://raw.githubusercontent.com/rh-nfv-int/shift-on-stack-vhostuser/master/roles/sos-vhostuser/files/vhostuser | base64 -w 0)
```

Comment:
Should we copy the script locally?

Reply:
I told the maintainer that we were using it so we need it to be "stable". We'll see if that breaks, but I think it's nice to get latest version, as you could see we improved it a bit lately.